histexpand: add page (#5455)
tldr-pages/tldr: pages/common/zsh.md

# zsh

> > Z SHell, a `bash` and `sh`-compatible command line interpreter.
> > Please refer to `tldr histexpand` for history expansion.

Reviewer: Perhaps this will read better?
```suggestion
> Z SHell, a `bash` and `sh`-compatible command line interpreter.
> See also `histexpand` for history expansion.
```
